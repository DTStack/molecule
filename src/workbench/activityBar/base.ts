import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';
import { ID_ACTIVITY_BAR } from 'mo/common/id';

export const defaultClassName = prefixClaName(ID_ACTIVITY_BAR);
export const containerClassName = getBEMElement(defaultClassName, 'container');
export const normalItemsClassName = getBEMElement(defaultClassName, 'normal');
export const globalItemsClassName = getBEMElement(defaultClassName, 'global');
export const itemClassName = getBEMElement(defaultClassName, 'item');
export const itemCheckedClassName = getBEMModifier(itemClassName, 'checked');
export const itemDisabledClassName = getBEMModifier(itemClassName, 'disabled');
export const labelClassName = getBEMElement(defaultClassName, 'label');
export const indicatorClassName = getBEMElement(defaultClassName, 'indicator');
