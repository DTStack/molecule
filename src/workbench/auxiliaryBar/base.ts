import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';

export const defaultClassName = prefixClaName('auxiliaryBar');
export const containerClassName = getBEMElement(defaultClassName, 'container');

export const tabsClassName = getBEMElement(defaultClassName, 'tabs');
export const tabClassName = getBEMElement(tabsClassName, 'tab');

export const tabActiveClassName = getBEMModifier(tabClassName, 'active');
