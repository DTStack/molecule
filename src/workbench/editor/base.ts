import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';

export const defaultEditorClassName = prefixClaName('editor');
export const groupClassName = getBEMElement(defaultEditorClassName, 'group');
export const groupContainerClassName = getBEMElement(
    defaultEditorClassName,
    'group-container'
);
export const groupHeaderClassName = getBEMElement(
    defaultEditorClassName,
    'group-header'
);
export const groupTabsClassName = getBEMElement(
    defaultEditorClassName,
    'group-tabs'
);
export const groupActionsClassName = getBEMElement(
    defaultEditorClassName,
    'group-actions'
);
export const groupActionsItemClassName = getBEMElement(
    defaultEditorClassName,
    'group-actions-item'
);

export const groupActionItemDisabledClassName = getBEMModifier(
    groupActionsItemClassName,
    'disabled'
);

export const groupBreadcrumbClassName = getBEMElement(
    defaultEditorClassName,
    'group-breadcrumb'
);
