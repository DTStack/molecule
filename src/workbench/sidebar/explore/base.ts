import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';
import {
    ID_ACTIVITY_BAR,
    ID_SIDE_BAR,
    ID_EXPLORER,
    ID_FOLDER_TREE,
    ID_EDITOR_TREE,
} from 'mo/common/id';

const defaultClassName = prefixClaName(ID_SIDE_BAR);
const defaultExplorerClassName = prefixClaName(ID_EXPLORER, defaultClassName);
const activityBarItemFloatClassName = getBEMModifier(
    getBEMElement(defaultExplorerClassName, ID_ACTIVITY_BAR),
    'float'
);

const folderTreeClassName = prefixClaName(ID_FOLDER_TREE);
const folderTreeInputClassName = getBEMModifier(folderTreeClassName, 'input');
const folderTreeEditClassName = getBEMModifier(folderTreeClassName, 'editable');

const editorTreeClassName = prefixClaName(ID_EDITOR_TREE);
const editorTreeItemClassName = getBEMElement(editorTreeClassName, 'item');
const editorTreeGroupClassName = getBEMElement(editorTreeClassName, 'group');
const editorTreeFileNameClassName = getBEMElement(
    editorTreeItemClassName,
    'fileName'
);
const editorTreeFilePathClassName = getBEMElement(
    editorTreeItemClassName,
    'filePath'
);
const editorTreeActiveItemClassName = getBEMModifier(
    editorTreeItemClassName,
    'active'
);
const editorTreeCloseIconClassName = getBEMElement(
    editorTreeClassName,
    'close'
);
const editorTreeFileIconClassName = getBEMElement(editorTreeClassName, 'file');

export {
    defaultExplorerClassName,
    activityBarItemFloatClassName,
    folderTreeClassName,
    folderTreeInputClassName,
    folderTreeEditClassName,
    editorTreeClassName,
    editorTreeItemClassName,
    editorTreeGroupClassName,
    editorTreeFileNameClassName,
    editorTreeFilePathClassName,
    editorTreeActiveItemClassName,
    editorTreeCloseIconClassName,
    editorTreeFileIconClassName,
};
