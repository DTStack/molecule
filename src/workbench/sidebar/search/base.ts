import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';

const emptyTextValueClassName = getBEMModifier(
    getBEMElement(prefixClaName('search'), 'treeNode'),
    'empty'
);

const matchSearchValueClassName = getBEMModifier(
    getBEMElement(prefixClaName('search'), 'treeNode'),
    'match'
);

const deleteSearchValueClassName = getBEMModifier(
    getBEMElement(prefixClaName('search'), 'treeNode'),
    'delete'
);

const replaceSearchValueClassName = getBEMModifier(
    getBEMElement(prefixClaName('search'), 'treeNode'),
    'replace'
);

const treeContentClassName = getBEMElement(prefixClaName('search'), 'tree');

export {
    matchSearchValueClassName,
    emptyTextValueClassName,
    deleteSearchValueClassName,
    replaceSearchValueClassName,
    treeContentClassName,
};
