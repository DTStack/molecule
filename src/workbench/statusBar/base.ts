import { getBEMElement, prefixClaName } from 'mo/common/className';
import { IStatusBarItem } from 'mo/model';

export const statusBarClassName = prefixClaName('statusBar');
export const leftItemsClassName = getBEMElement(
    statusBarClassName,
    'left-items'
);
export const rightItemsClassName = getBEMElement(
    statusBarClassName,
    'right-items'
);
export const itemClassName = getBEMElement(statusBarClassName, 'item');

export function sortByIndex(a: IStatusBarItem, b: IStatusBarItem) {
    return a.sortIndex !== undefined && b.sortIndex !== undefined
        ? a.sortIndex - b.sortIndex
        : 0;
}
