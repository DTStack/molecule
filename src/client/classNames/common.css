*:focus-visible {
    outline: none;
}

body [tabindex='0']:focus,
body [tabindex='-1']:focus,
body .synthetic-focus,
body select:focus,
body input[type='button']:focus,
body input[type='text']:focus,
body button:focus,
body textarea:focus,
body input[type='search']:focus,
body input[type='checkbox']:focus {
    outline-width: 1px;
    outline-style: solid;
    outline-offset: -1px;
    outline-color: var(--focusBorder);
    opacity: 1;
}

body .synthetic-focus :focus {
    outline: 0 !important; /* elements within widgets that draw synthetic-focus should never show focus */
}

body .monaco-inputbox.info.synthetic-focus,
body .monaco-inputbox.warning.synthetic-focus,
body .monaco-inputbox.error.synthetic-focus,
body .monaco-inputbox.info input[type='text']:focus,
body .monaco-inputbox.warning input[type='text']:focus,
body .monaco-inputbox.error input[type='text']:focus {
    outline: 0 !important; /* outline is not going well with decoration */
}

body .monaco-list:focus {
    outline: 0 !important; /* tree indicates focus not via outline but through the focused item */
}

body {
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial,
        sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji';
}
