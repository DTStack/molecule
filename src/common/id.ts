export const ID_APP = 'molecule';
export const ID_ACTIVITY_BAR = 'activityBar';
export const ID_MENU_BAR = 'menuBar';
export const ID_SIDE_BAR = 'sidebar';
export const ID_EXPLORER = 'explorer';
