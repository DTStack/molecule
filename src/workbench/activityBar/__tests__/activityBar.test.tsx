import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ID_ACTIVITY_BAR } from 'mo/common/id';
import type { IActivityBarItem } from 'mo/workbench';
import { itemCheckedClassName, itemClassName } from '../base';
import ActivityBar from '../activityBar';

const mockData: IActivityBarItem[] = [
    {
        id: 'test',
        name: 'test',
        icon: 'account',
        type: 'global',
    },
    {
        id: 'test2',
        name: 'test2',
        icon: 'settings-gear',
        type: 'global',
    },
    {
        id: 'test3',
        name: 'test3',
        icon: 'debug',
    },
];

describe('The ActivityBar Component', () => {
    test('match the snapshot', () => {
        let component;
        act(() => {
            component = renderer.create(<ActivityBar data={mockData} />);
        });
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Should support to specify the selected one', () => {
        const { container } = render(
            <ActivityBar data={mockData} selected="test3" />
        );

        const selectedOne = container.querySelector('#test3');
        expect(selectedOne?.classList).toContain(itemCheckedClassName);

        const unselected = container.querySelector('#test');
        expect(unselected?.classList).not.toContain(itemCheckedClassName);
    });

    test('Should trigger onClick event and onChange event', () => {
        const mockClickFn = jest.fn();
        const mockChangeFn = jest.fn();
        const { container } = render(
            <ActivityBar
                data={mockData}
                onClick={mockClickFn}
                onChange={mockChangeFn}
            />
        );

        const item = container.querySelector('#test3')!;
        const globalItem = container.querySelector('#test')!;

        fireEvent.click(item);

        expect(mockChangeFn).toBeCalledTimes(1);
        expect(mockClickFn).toBeCalledTimes(1);

        fireEvent.click(globalItem);

        // global item won't trigger onChange
        expect(mockChangeFn).toBeCalledTimes(1);
        expect(mockClickFn).toBeCalledTimes(2);

        expect(mockClickFn.mock.calls[0][0]).toBe('test3');
        expect(mockClickFn.mock.calls[0][1]).toEqual(
            expect.objectContaining({
                id: 'test3',
                name: 'test3',
                icon: 'debug',
            })
        );

        expect(mockChangeFn.mock.calls[0][0]).toBeUndefined();
        expect(mockChangeFn.mock.calls[0][1]).toBe('test3');
    });

    test('Should support to show contextMenu when not above item', () => {
        const originalFunc = document.elementsFromPoint;
        const mockFn = jest.fn();
        const contextMenus = [
            {
                id: 'menus',
                name: 'menus',
                icon: 'check',
            },
        ];
        const { container, getByRole } = render(
            <ActivityBar
                data={mockData}
                contextMenu={contextMenus}
                onContextMenuClick={mockFn}
            />
        );

        document.elementsFromPoint = jest.fn(() => {
            return [document.createElement('div')];
        });

        fireEvent.contextMenu(container.querySelector(`#${ID_ACTIVITY_BAR}`)!);

        const menus = getByRole('menu');
        expect(menus).toBeInTheDocument();

        fireEvent.click(menus.firstElementChild!);
        expect(mockFn).toBeCalled();
        expect(mockFn.mock.calls[0][1]).toEqual(
            expect.objectContaining(contextMenus[0])
        );

        document.elementsFromPoint = originalFunc;
    });

    test('Should show different contextMenu when rightClick above item', () => {
        const originalFunc = document.elementsFromPoint;
        const contextMenus = [
            {
                id: 'menus',
                name: 'menus',
                icon: 'check',
            },
        ];
        const { container, getByRole } = render(
            <ActivityBar data={mockData} contextMenu={contextMenus} />
        );

        // mock elementsFromPoint to make ensure current cursor above the item
        document.elementsFromPoint = jest.fn(() => {
            const div = document.createElement('div');
            div.id = 'menus';
            div.classList.add(itemClassName);
            return [div];
        });

        fireEvent.contextMenu(container.querySelector(`#${ID_ACTIVITY_BAR}`)!);

        const menus = getByRole('menu');
        expect(menus).toBeInTheDocument();

        expect(menus.firstElementChild).not.toBeNull();
        expect(
            menus.firstElementChild!.querySelector('.codicon-check')
        ).toBeInTheDocument();
        expect(getByRole('separator')).toBeInTheDocument();

        document.elementsFromPoint = originalFunc;
    });
});
