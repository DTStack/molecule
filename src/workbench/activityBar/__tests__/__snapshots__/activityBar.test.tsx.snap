// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ActivityBar Component match the snapshot 1`] = `
<div
  className="mo-activityBar"
  id="activityBar"
  onContextMenu={[Function]}
>
  <div
    className="mo-activityBar__container"
  >
    <ul>
      <li
        className="mo-activityBar__item"
        id="test3"
        onClick={[Function]}
      >
        <span
          className="mo-activityBar__label codicon codicon-debug"
          title=""
        />
      </li>
    </ul>
    <ul
      className="mo-activityBar__global"
    >
      <li
        className="mo-activityBar__item"
        id="test"
        onClick={[Function]}
      >
        <span
          className="mo-activityBar__label codicon codicon-account"
          title=""
        />
      </li>
      <li
        className="mo-activityBar__item"
        id="test2"
        onClick={[Function]}
      >
        <span
          className="mo-activityBar__label codicon codicon-settings-gear"
          title=""
        />
      </li>
    </ul>
  </div>
</div>
`;
